src/widgets/toggleRefinement/toggleRefinement.js

```diff
@@ -0,0 +1,168 @@
+ * @property {boolean} collapsed If set to true, the widget will be collapsed at first rendering.
+ */
+
+/**
+ * @typedef {Object} ToggleWidgetOptions
+ * @property {string|HTMLElement} container Place where to insert the widget in your webpage.
+ * @property {string} attribute Name of the attribute for faceting (eg. "free_shipping").
+ * @property  {string|number|boolean} on Value to filter on when checked.
+ * @property  {string|number|boolean} off Value to filter on when unchecked.
+ * element (when using the default template). By default when switching to `off`, no refinement will be asked. So you
+ * will get both `true` and `false` results. If you set the off value to `false` then you will get only objects
+ * having `false` has a value for the selected attribute.
+ * @property {ToggleWidgetTemplates} [templates] Templates to use for the widget.
+ * @property {ToggleWidgetTransforms} [transformData] Object that contains the functions to be applied on the data * before being used for templating. Valid keys are `body` for the body template.
+ * @property {ToggleWidgetCSSClasses} [cssClasses] CSS classes to add.
+ */
+
+/**
+ * The toggleRefinement widget lets the user either:
+ *  - switch between two values for a single facetted attribute (free_shipping / not_free_shipping)
+ *  - toggleRefinement a faceted value on and off (only 'canon' for brands)
+ *
+ * This widget is particularly useful if you have a boolean value in the records.
+ *
+ * @requirements
+ * The attribute passed to `attribute` must be declared as an
+ * [attribute for faceting](https://www.algolia.com/doc/guides/searching/faceting/#declaring-attributes-for-faceting)
+ * in your Algolia settings.
+ *
+ * @type {WidgetFactory}
+ * @devNovel ToggleRefinement
+ * @category filter
+ * @param {ToggleWidgetOptions} $0 Options for the ToggleRefinement widget.
+ * @return {Widget} A new instance of the ToggleRefinement widget
+ * @example
+ * search.addWidget(
+ *   instantsearch.widgets.toggleRefinement({
+ *     container: '#free-shipping',
+ *     attribute: 'free_shipping',
+ *     on: true,
+ *     templates: {
+ *       labelText: 'Free shipping'
+ *     }
+ *   })
+ * );
+ */
+export default function toggleRefinement({
+  container,
+  attribute,
+  cssClasses: userCssClasses = {},
+  templates = defaultTemplates,
+  transformData,
+  on = true,
+  off,
+} = {}) {
+  if (!container) {
+    throw new Error(usage);
+  }
+
+  const containerNode = getContainerNode(container);
+
+  const cssClasses = {
+    root: cx(suit(), userCssClasses.root),
+    label: cx(suit({ descendantName: 'label' }), userCssClasses.label),
+    checkbox: cx(suit({ descendantName: 'checkbox' }), userCssClasses.checkbox),
+    labelText: cx(
+      suit({ descendantName: 'labelText' }),
+      userCssClasses.labelText
+    ),
+  };
+
+  const specializedRenderer = renderer({
+    containerNode,
+    cssClasses,
+    renderState: {},
+    templates,
+    transformData,
```

Comment:
We would like to drop it since it can be fully replaced by `transformItems`. We haven't removed it yet in other widgets either. We can do it in another PR.

Reply:
Yep, should be dropped IMO :)